### DISCUSSION TITLE
Feature Request: Support GROUPING SETS in PIVOT statements

### DISCUSSION THREAD

**Discussion by chanon** [duckdb/duckdb#12860] [Ideas]
It would be very nice if `PIVOT` statements supported `GROUPING SETS` in the `GROUP BY` clause so that we could for example get bottom-line overall totals and subtotals in a single `PIVOT` query.

Kind of related, but not the same thing, it would also be awesome if there was a way to get right side totals at the same time too.

**Comment by chanon** [comment]
Note that this would also cover:
https://github.com/duckdb/duckdb/discussions/12673

As then you could do `PIVOT .... ON .... USING ... GROUP BY GROUPING SETS (());`
With the added benefit of being able to also get grouped results at the same time too.

Eg, using the `Cities` example table:
```
PIVOT Cities 
ON Year 
USING sum(Population) 
GROUP BY GROUPING SETS ( (Country), () );`
```
Or even
```
PIVOT Cities 
ON Year 
USING sum(Population) 
GROUP BY ROLLUP (Country)`
```

**Comment by naxels** [comment]
This is exactly what I'm looking for :)
Just created a query with a GROUP BY ROLLUP and then discovered I could make it much simpler with PIVOT however lose the totals column and row..
